### PR TITLE
Clean up code, list containers

### LINKED ISSUE #21: why the `__ALL__` definitions ?
just want to outline that

* `__ALL__` should be spelled `__all__`, like all other special names
* plus, and more to the point, there are lighter conventions to just rely on the *public*  / *_private* naming convention, which seems to be in place already since `_report_not_found` follows the *_private* pattern

So I just wanted to make sure that it was mandatory to have these defined for some higher level reason, like it's a redhat policy or something

Otherwise I'd suggest to simply drop it, as it serves no visible purpose (evidence of that being that it is currently misspelled), and just adds possibility for error

### PR DESCRIPTION
**Pull request by jwhonce**
* Reformat project with black formatter/style + isort
* Rename `__ALL__` to `__all__`
* Implement list_containers() + unit tests

Fixes #21 

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
@mwhahaha PTAL

File: podman/tests/unit/test_api_connection.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def test_post(self):
                       headers={}),
             mock.call('POST', '/test/bar', body='a=b&c=d',
                       headers={
-                          'x': 'y',
+                          'x':            'y',
```
Comment (mwhahaha) [inline podman/tests/unit/test_api_connection.py:117]: Was this a magical formatter thing?

**Review by mwhahaha — APPROVED** [review]
looks fine

File: podman/containers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""containers provides the operations against containers for a Podman service.
+"""
+
+import json
+
+
+def list_containers(api, all=None):
```
Comment (mwhahaha) [inline podman/containers/__init__.py:7]: actually all is a built-in in python

File: podman/tests/unit/containers/test_containers.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    def test_list_containers(self):
+        """test list call"""
+        mock_read = mock.MagicMock()
+        mock_read.return_value = b'[{"Id": "foo"}]'
+        self.response.status = 200
+        self.response.read = mock_read
+        expected = [{"Id": "foo"}]
+        ret = podman.containers.list_containers(self.api)
+        self.assertEqual(ret, expected)
+        self.request.assert_called_once_with('/containers/json')
```
Comment (mwhahaha) [inline podman/tests/unit/containers/test_containers.py:34]: This is failing (i need to fix the makefile as it's not exiting on failure anymore). 

```
AssertionError: Expected call: get('/containers/json')
Actual call: get('/containers/json', {})
```

**Review by mwhahaha — CHANGES_REQUESTED** [review]

File: podman/tests/unit/test_api_connection.py (hunk 1 above)
Comment (jwhonce) [reply podman/tests/unit/test_api_connection.py:117]: Yeah, not my favorite.

File: podman/images/__init__.py [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
     """List all images for a Podman service."""
     response = api.get("/images/json")
-    return json.loads(str(response.read(), 'utf-8'))
+    return json.loads(str(response.read(), "utf-8"))
 
 
 def inspect(api, name):
     """Report on named image for a Podman service.
-       Name may also be a image ID.
+    Name may also be a image ID.
```
Comment (TomSweeneyRedHat) [inline podman/images/__init__.py:16]: nit of a nit, only change if you've other tweaks.
```suggestion
    Name may also be an image ID.
```

**Comment by jwhonce** [comment]
@mwhahaha: r+

File: podman/images/__init__.py (hunk 4 above)
Comment (TomSweeneyRedHat) [reply podman/images/__init__.py:16]: Actually, this is user-facing, can you clean this and I'll LGTM it?

File: podman/api_connection.py [hunk 5]
```diff
@@ -7,26 +7,31 @@
...
 class ApiConnection(HTTPConnection, AbstractContextManager):
-    """ ApiConnection provides a specialized HTTPConnection
-        to a Podman service."""
+    """ApiConnection provides a specialized HTTPConnection
+    to a Podman service."""
 
-    def __init__(self, url, base="/v1.24/libpod", *args,
-                 **kwargs):  # pylint: disable-msg=W1113
+    def __init__(
+        self, url, base="/v1.24/libpod", *args, **kwargs
```
Comment (mwhahaha) [inline podman/api_connection.py:21]: pylint seems unhappy with this for whatever reason, but we can resolve that later in the black usage patch.

**Review by mwhahaha — APPROVED** [review]
LGTM

**Pull request merged**